pkg/controller/resolvedofferings/computeclass.go

```diff
@@ -45,19 +48,24 @@ func resolveComputeClasses(req router.Request, cfg *apiv1.Config, appInstance *v
 			return err
 		}
 		def := parsedMemory.Def.Value()
+		cpuQuantity := computeclasses.CalculateCPU(*cc, *parsedMemory.Def)
 		appInstance.Status.ResolvedOfferings.Containers[""] = v1.ContainerResolvedOffering{
-			Memory:    &def,
-			CPUScaler: &cc.CPUScaler,
-			Class:     appInstance.Status.ResolvedOfferings.Containers[""].Class,
+			Memory: &def,
+			CPU:    z.Pointer(cpuQuantity.MilliValue()),
+			Class:  appInstance.Status.ResolvedOfferings.Containers[""].Class,
 		}
 	}
 
 	// Check to see if the user overrode the memory for all containers
-	if appInstance.Spec.Memory[""] != nil {
+	if specificMemory := appInstance.Spec.Memory[""]; specificMemory != nil {
+		var cpuQuantity resource.Quantity
+		if cc != nil {
+			cpuQuantity = computeclasses.CalculateCPU(*cc, *resource.NewQuantity(*specificMemory, resource.BinarySI))
+		}
 		appInstance.Status.ResolvedOfferings.Containers[""] = v1.ContainerResolvedOffering{
-			Memory:    appInstance.Spec.Memory[""],
-			CPUScaler: appInstance.Status.ResolvedOfferings.Containers[""].CPUScaler,
-			Class:     appInstance.Status.ResolvedOfferings.Containers[""].Class,
+			Memory: appInstance.Spec.Memory[""],
+			CPU:    z.Pointer(cpuQuantity.MilliValue()),
```

Comment:
just curious: is there a behavioral difference between `nil` and a pointer to a zero value?

Reply:
Talked offline. This reflects a weirder issue in quota. I'm going to make an issue for that, merge this, and come back to fix what this points to.